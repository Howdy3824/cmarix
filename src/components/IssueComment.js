import React, { PropTypes } from 'react';
import ReactMarkdown from 'react-markdown';
import UserWithAvatar from './UserWithAvatar';
import { insertMentionLinks } from '../utils/stringUtils';

export default function IssueComment({ comment }) {
  return (
    <div className="issue-detail__comment">
      <UserWithAvatar user={comment.user} orientation="horizontal"/>

      <div className="issue-detail__comment__body">
        <ReactMarkdown className="markdown" source={insertMentionLinks(comment.body)}/>
      </div>
    </div>
  );
}

IssueComment.propTypes = {
  comment: PropTypes.shape({
    user: PropTypes.shape({
      login: PropTypes.string,
      avatar_url: PropTypes.string
    }).isRequired,
    body: PropTypes.string
  }).isRequired
};