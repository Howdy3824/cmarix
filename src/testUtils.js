import toJson from 'enzyme-to-json';

export function afterPromises(done, fn) {
  setTimeout(() => {
    try {
      fn();
      done();
    } catch (e) {
      fail();
    }
  }, 1);
}

export function checkSnapshot(tree) {
  expect(toJson(tree)).toMatchSnapshot();
}