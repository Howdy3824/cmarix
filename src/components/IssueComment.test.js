import React from 'react';
import { shallow } from 'enzyme';
import { checkSnapshot } from '../utils/testUtils';
import IssueComment from './IssueComment';

it('renders a comment', () => {
  const comment = {
    user: {
      login: "somebody",
      avatar_url: "http://foo"
    },
    body: "a comment"
  };
  const tree = shallow(
    <IssueComment comment={comment}/>
  );
  checkSnapshot(tree);
});