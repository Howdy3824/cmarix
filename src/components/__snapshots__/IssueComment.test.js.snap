exports[`test renders a comment 1`] = `
<div
  className="issue-detail__comment">
  <UserWithAvatar
    orientation="horizontal"
    user={
      Object {
        "avatar_url": "http://foo",
        "login": "somebody",
      }
    } />
  <div
    className="issue-detail__comment__body">
    <ReactMarkdown
      className="markdown"
      containerTagName="div"
      source="a comment" />
  </div>
</div>
`;
